lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -614,6 +624,91 @@ export function detectTestingLibraryUtils<
       );
     };
 
+    /**
+     * Determines whether a given node is some reportable `act` util.
+     *
+     * An `act` is reportable if some of these conditions is met:
+     * - it's related to Testing Library module (this depends on Aggressive Reporting)
+     * - it's related to React DOM Test Utils
+     */
+    const isActUtil = (node: TSESTree.Identifier): boolean => {
+      const isTestingLibraryAct = isPotentialTestingLibraryFunction(
+        node,
+        (identifierNodeName, originalNodeName) => {
+          return [identifierNodeName, originalNodeName]
+            .filter(Boolean)
+            .includes('act');
+        }
+      );
+
+      const isReactDomTestUtilsAct = (() => {
```

Comment:
I know there are several IIFE used here and some blocks could be refactored or extracted. I'm gonna leave it like this so first I fix the issues found and then, having proper context, I'll refactor and extract corresponding functions.